optimum/fx/optimization/transformations.py

```diff
@@ -293,15 +293,64 @@ def transform(self, graph_module: "GraphModule") -> "GraphModule":
         return graph_module
 
     def reverse(self, graph_module: "GraphModule") -> "GraphModule":
-        named_modules = dict(graph_module.named_modules())
         for node in graph_module.graph.nodes:
             if node.op == "call_module":
                 if getattr(node, "was_transformed", "") == "MergeLinears":
-                    self._unmerge_linears(graph_module, node, named_modules[node.target])
+                    self._unmerge_linears(graph_module, node, graph_module.get_submodule(node.target))
 
         return graph_module
 
 
+@add_docstring()
+class FuseBiasInLinear(ReversibleTransformation):
+    """
+    Transformation that fuses the bias to the weight in torch.nn.Linear.
+    """
+
+    preserves_computation = True
+
+    def transform(self, graph_module: "GraphModule") -> "GraphModule":
+        torch_ones = _gen_constructor_wrapper(torch.ones)[0]
+
+        def insert_concat(linear_input):
+            shape = linear_input.shape[:-1] + (1,)
+            return torch.cat([linear_input, torch_ones(shape)], dim=-1)
+
+        tracer = torch.fx.proxy.GraphAppendingTracer(graph_module.graph)
+        for node in graph_module.graph.nodes:
+            if node.op == "call_module":
+                module = graph_module.get_submodule(node.target)
+                if isinstance(module, torch.nn.Linear) and module.bias is not None:
+                    with graph_module.graph.inserting_before(node):
+                        linear_input_proxy = torch.fx.Proxy(node.args[0], tracer)
+                        output_proxy = insert_concat(linear_input_proxy)
+                        node.original_node_args = node.args
+                        node.args = (output_proxy.node,)
+                        node.was_transformed = "FuseBiasInLinear"
+
+                    new_weight = torch.nn.Parameter(torch.cat([module.weight, module.bias[:, None]], dim=1))
+                    module.weight = new_weight
+                    module.bias = None
+        return graph_module
+
+    def reverse(self, graph_module: "GraphModule") -> "GraphModule":
+        for node in graph_module.graph.nodes:
+            if getattr(node, "was_transformed", "") == "FuseBiasInLinear":
+                inserted_nodes = [node.args[0]]
+                node.args = node.original_node_args
+                while inserted_nodes:
+                    n = inserted_nodes.pop(0)
+                    if n is not node.original_node_args[0]:
```

Comment:
When could this happen? When some linear layers have been merged?

Reply:
So this idea here is to delete all the inserted nodes by the `insert_concat` function in the `transform` method. 
This fuctions "starts" from `node.original_node_args[0]`, and the plan is to remove the nodes between this one and the output node. 

But while explaining it to you, I noticed that it is not what's happening here... thanks for catching this!